compiler/rustc_query_impl/src/plumbing.rs

```diff
@@ -457,20 +454,7 @@ macro_rules! define_queries {
                 }
 
                 fn force_from_dep_node(tcx: QueryCtxt<'_>, dep_node: &DepNode) -> bool {
-                    if is_anon {
-                        return false;
-                    }
-
-                    if !can_reconstruct_query_key() {
-                        return false;
-                    }
-
-                    if let Some(key) = recover(*tcx, dep_node) {
-                        force_query::<queries::$name<'_>, _>(tcx, key, DUMMY_SP, *dep_node);
-                        return true;
-                    }
-
-                    false
+                    force_query::<queries::$name<'_>, _>(tcx, dep_node)
```

Comment:
Can we directly take a function pointer to this rather than (hoping) that LLVM sees through this layer of indirection?

Reply:
I get a higher-ranked subtype error when trying to take a function pointer.

Reply:
Hm, that seems unfortunate. Feel free to re-run perf and/or r=me as I said.